crates/bevy_pbr/src/render/mesh.rs

```diff
@@ -508,27 +645,42 @@ impl FromWorld for MeshPipeline {
             entries.extend_from_slice(&tonemapping_lut_entries);
 
             if cfg!(any(not(feature = "webgl"), not(target_arch = "wasm32")))
-                || (cfg!(all(feature = "webgl", target_arch = "wasm32")) && !multisampled)
+                || (cfg!(all(feature = "webgl", target_arch = "wasm32"))
+                    && !layout_key.contains(MeshPipelineViewLayoutKey::MULTISAMPLED))
             {
                 entries.extend_from_slice(&prepass::get_bind_group_layout_entries(
                     [17, 18, 19, 20],
-                    multisampled,
+                    layout_key,
                 ));
             }
 
             entries
         }
 
-        let view_layout = render_device.create_bind_group_layout(&BindGroupLayoutDescriptor {
-            label: Some("mesh_view_layout"),
-            entries: &layout_entries(clustered_forward_buffer_binding_type, false),
-        });
+        let mut i: u32 = 0;
+        let view_layouts = [(); MeshPipelineViewLayoutKey::COUNT].map(|_| {
```

Comment:
Consider using [`array::from_fn`](https://doc.rust-lang.org/stable/std/array/fn.from_fn.html) here.

Reply:
Didn't know about this one, it's indeed much cleaner.